cmd/beekeeper/cmd/cluster.go

```diff
@@ -0,0 +1,230 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/config"
+	"golang.org/x/sync/errgroup"
+)
+
+func (c *command) deleteCluster(ctx context.Context, clusterName string, cfg *config.Config, deleteStorage bool) (err error) {
+	clusterConfig, ok := cfg.Clusters[clusterName]
+	if !ok {
+		return fmt.Errorf("cluster %s not defined", clusterName)
+	}
+
+	clusterOptions := clusterConfig.Export()
+	clusterOptions.K8SClient = c.k8sClient
+
+	cluster := bee.NewCluster(clusterConfig.GetName(), clusterOptions)
+
+	// delete node groups
+	for ng, v := range clusterConfig.GetNodeGroups() {
+		fmt.Printf("deleting %s node group\n", ng)
+		ngConfig, ok := cfg.NodeGroups[v.Config]
+		if !ok {
+			return fmt.Errorf("node group profile %s not defined", v.Config)
+		}
+		if v.Mode == "bootnode" { // TODO: implement standalone mode
+			// register node group
+			cluster.AddNodeGroup(ng, ngConfig.Export())
+
+			// delete nodes from the node group
+			g := cluster.NodeGroup(ng)
+			for i := 0; i < len(v.Nodes); i++ {
+				nName := v.Nodes[i].Name
+				if err := g.DeleteNode(ctx, nName); err != nil {
+					return fmt.Errorf("deleting node %s from the node group %s", nName, ng)
+				}
+
+				if deleteStorage && *ngConfig.PersistenceEnabled {
+					pvcName := fmt.Sprintf("data-%s-0", nName)
+					if err := c.k8sClient.PVC.Delete(ctx, pvcName, clusterOptions.Namespace); err != nil {
+						return fmt.Errorf("deleting pvc %s: %w", pvcName, err)
+					}
+				}
+			}
```

Comment:
Shouldn't return nil. Else can be removed, but I would keep it for now as new mode will be implemented soon. This way it will be clear what are whole block and how optimisation should be done.